File: ruby/splitwise/lib/splitwise/models/user_settlement.rb
module Splitwise
  module Models
    class UserSettlement
      attr_accessor :payer, :payee, :amount

      def initialize(payer:, payee:, amount:)
        self.payer = payer                   # who pays
        self.payee = payee                   # who gets paid
        self.amount = amount                 # the amount
      end

      def to_s
        "#{payer.name} pays #{payee.name} Rs. #{amount}"
      end
    end
  end
end

File: java/splitwise/src/main/java/gagan93jtg/lld/models/UserSettlement.java
package gagan93jtg.lld.models;

public class UserSettlement {
  String payer; // Who pays
  String payee; // Who gets paid
  double amount;

  public UserSettlement(String payer, String payee, double amount) {
    this.payer = payer;
    this.payee = payee;
    this.amount = amount;
  }

  public String getPayer() {
    return payer;
  }

  public String getPayee() {
    return payee;
  }

  public double getAmount() {
    return amount;
  }

  @Override
  public String toString() {
    return payer + " pays " + payee + " Rs. " + amount ;
  }
}
